### PR TITLE
Offline quizzes

### PR DESCRIPTION
**Pull request by szabinst**
refs: MBL-16768
affects: Student
release note: none 
test plan: 
1. Select quizzes when syncing offline content 
2. Turn off internet access
3. Open Quizzes tab
4. You should be able to see the quiz list 
5. On the details page, the `Take quiz`, `Resume quiz`, `View result` button should be disabled. 

## Checklist

- [x] Tested on phone
- [x] Tested on tablet

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-16768](https://instructure.atlassian.net/browse/MBL-16768)

<details><summary>:x: XCTest failed: CoreTests/CourseSyncSelectorInteractorLiveTests/testTabList</summary>


    XCTAssertEqual failed: ("4") is not equal to ("3")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 09f8bb9f7695805f4a0e22f29180f1263ca60182
</p>

File: Core/Core/CourseSync/CourseSyncDownloader/Model/CourseSyncQuizzesInteractor.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        .map { _ in () }
+        .eraseToAnyPublisher()
+    }
+
+    private func getCustomColors(courseId _: String) -> AnyPublisher<Void, Error> {
+        ReactiveStore(
+            useCase: GetCustomColors()
+        )
+        .getEntities()
+        .map { _ in () }
```
Comment (vargaat) [inline Core/Core/CourseSync/CourseSyncDownloader/Model/CourseSyncQuizzesInteractor.swift:44]: `CombineExt` provides a nice `.mapToVoid()` function for this.

**Review by vargaat — COMMENTED** [review]
Code quality looks good. Sync fails for new quizzes. Also the new quiz's "Launch External Tool" button is enabled in offline mode.

**Comment by szabinst** [comment]
Modified the download logic so it filters out new quizzes and LTI button is now also hidden.

**Comment by szabinst** [comment]
@balintbartok I pushed a fix where this type of quiz is covered.

File: Student/Student/Quizzes/QuizDetails/QuizDetailsViewController.swift [hunk 2]
```diff
@@ -162,6 +162,7 @@ class QuizDetailsViewController: ScreenViewTrackableViewController, ColoredNavVi
         let title = takeButtonTitle
         takeButton.setTitle(title, for: .normal)
         takeButton.isHidden = title == nil
+        takeButton.makeUnavailableInOfflineMode()
```
Comment (idevainstr) [inline Student/Student/Quizzes/QuizDetails/QuizDetailsViewController.swift:165]: The takeButton gets funky when this is applied, we might need to change its z order either in code or on the storyboard.
IMG_5A058E9D8721-1

**Review by idevainstr — CHANGES_REQUESTED** [review]

**Review by balintbartok — DISMISSED** [review]
QA + 1

**Review by ndrsszsz — APPROVED** [review]
QA +1

**Pull request merged**